src/sdam/description/topology/mod.rs

```diff
@@ -849,10 +871,12 @@ impl Default for TransactionSupportStatus {
 /// Returned from `TopologyDescription::diff`.
 #[derive(Debug)]
 pub(crate) struct TopologyDescriptionDiff<'a> {
-    pub(crate) removed_addresses: HashSet<&'a ServerAddress>,
-    pub(crate) added_addresses: HashSet<&'a ServerAddress>,
-    pub(crate) changed_servers:
-        HashMap<&'a ServerAddress, (&'a ServerDescription, &'a ServerDescription)>,
+    pub(crate) removed_addresses: Vec<&'a ServerAddress>,
```

Comment:
Is the change to `Vec` purely to allow sorting in tests?  It looks like they're still "morally" sets, i.e. there won't be any duplicate keys.  If so, it might be clearer to do the sorting when emitting the events in `TopologyWorker::process_topology_diff` and leave these as `HashSet/Map`.

Reply:
Good idea, done.